neo/Prompt/Commands/Tokens.py

```diff
@@ -77,23 +80,103 @@ def command_desc(self):
 
 
-def token_send(wallet, args, prompt_passwd=True):
-    if len(args) < 4:
-        print("please provide a token symbol, from address, to address, and amount")
-        return False
+class CommandTokenSend(CommandBase):
+
+    def __init__(self):
+        super().__init__()
 
-    user_tx_attributes = None
-    if len(args) > 4:
-        args, user_tx_attributes = get_tx_attr_from_args(args)
+    def execute(self, arguments):
+        wallet = PromptData.Wallet
+
+        if len(arguments) < 4:
+            print("Please specify the required parameters")
+            return False
+
+        if len(arguments) > 5:
+            # the 5th argument is the optional attributes,
+            print("Too many parameters supplied. Please check your command")
+            return False
+
+        _, user_tx_attributes = get_tx_attr_from_args(arguments)
+
+        token = arguments[0]
+        send_from = arguments[1]
+        send_to = arguments[2]
+        try:
+            amount = float(arguments[3])
+        except ValueError:
+            print(f"{arguments[3]} is not a valid amount")
+            return False
+
+        try:
+            success = token_send(wallet, token, send_from, send_to, amount, user_tx_attributes)
+        except ValueError as e:
+            # occurs if arguments are invalid
+            print(str(e))
+            success = False
+
+        return success
+
+    def command_desc(self):
+        p1 = ParameterDesc('token', 'token symbol name or script_hash')
+        p2 = ParameterDesc('from_addr', 'address to send token from')
+        p3 = ParameterDesc('to_addr', 'address to send token to')
+        p4 = ParameterDesc('amount', 'number of tokens to send')
+        p5 = ParameterDesc('--tx-attr', f"a list of transaction attributes to attach to the transaction\n\n"
+        f"{' ':>17} See: http://docs.neo.org/en-us/network/network-protocol.html section 4 for a description of possible attributes\n\n"  # noqa: E128 ignore indentation
+        f"{' ':>17} Example:\n"
+        f"{' ':>20} --tx-attr=[{{\"usage\": <value>,\"data\":\"<remark>\"}}, ...]\n"
+        f"{' ':>20} --tx-attr=[{{\"usage\": 0x90,\"data\":\"my brief description\"}}]\n", optional=True)
+
+        return CommandDesc('send', 'send a token from the wallet', [p1, p2, p3, p4, p5])
+
+
+def token_send(wallet, token_str, send_from, send_to, amount, prompt_passwd=True, user_tx_attributes=None):
+    """
+
+    Args:
+        wallet (Wallet): a UserWallet instance
+        token_str (str): symbol name or script_hash
+        send_from (str): a wallet address
+        send_to (str): a wallet address
+        amount (float): the number of tokens to send
+        prompt_passwd (str): (optional) whether to prompt for a password before sending it to the network
+        user_tx_attributes (list): a list of ``TransactionAttribute``s.
+
+    Returns:
+        a Transaction object if successful, False otherwise.
```

Comment:
I think it's None if it fails

Reply:
My bad!
I checked something and it was returning `None`, I don't know what I checked 🤔